scripts/o3de/ExportScripts/export_source_built_project.py

```diff
@@ -0,0 +1,236 @@
+                                                 fail_on_asset_errors: bool = False,
+                                                 logger: logging.Logger|None = None):
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    default_base_build_path = ctx.engine_path / 'build' if ctx.is_engine_centric else ctx.project_path / 'build'
+
+    if not tools_build_path:
+        tools_build_path = default_base_build_path / 'tools'
+    if not game_build_path:
+        game_build_path = default_base_build_path / 'game'
+
+    exp.kill_existing_processes(ctx.project_name)
+
+    # Optionally build the toolchain needed to bundle the assets
+    if should_export_toolchain:
+        exp.build_export_toolchain(ctx=ctx,
+                                   tools_build_path=tools_build_path,
+                                   logger=logger)
+    else:
+        # Otherwise make sure the tools exist already
+        build_tools = [exp.get_asset_bundler_batch_path(tools_build_path), exp.get_asset_processor_batch_path(tools_build_path)]
+        tools_missing = [b for b in build_tools if not b.exists()]
+        if len(tools_missing) > 0:
+            raise exp.ExportProjectError(f"Necessary Build Tools have not been created! The following are missing: {', '.join(tools_missing)}"
+                                         "Please ensure that these tools exist before proceeding on with the build!")
+
+    if should_build_all_code:
+        exp.build_game_targets(ctx=ctx,
+                               build_config=build_config,
+                               game_build_path=game_build_path,
+                               should_build_game_launcher=should_build_game_launcher,
+                               should_build_server_launcher=should_build_server_launcher,
+                               should_build_unified_launcher=should_build_unified_launcher,
+                               allow_registry_overrides=allow_registry_overrides,
+                               logger=logger)
+
+    if should_build_all_assets:
+        exp.build_assets(ctx=ctx,
+                         tools_build_path=tools_build_path,
+                         fail_on_ap_errors=fail_on_asset_errors,
+                         logger=logger)
+
+        expected_bundles_path = exp.bundle_assets(ctx=ctx,
+                                                  selected_platform=selected_platform,
+                                                  seedlist_paths=seedlist_paths,
+                                                  non_mono_build_path=tools_build_path,
+                                                  custom_asset_list_path=preferred_asset_list_path,
+                                                  max_bundle_size=max_bundle_size)
+    else:
+        expected_bundles_path = ctx.project_path / "AssetBundling/Bundles"
+
+    for should_do, expected_output_dir, project_file_patterns, ignore_patterns in zip(
```

Comment:
I'm beginning to think the zipped objects warrant a dedicated class structure for the project export script. Something like this class definition:

```python
class SetupLayoutConfig:
   def __init__(self, output_path, should_build = True, project_file_patterns = [], ignore_patterns = [] ):
      self.should_build = should_build
      self.output_path = output_path
      self.project_file_patterns = project_file_patterns
      self.ignore_patterns = ignore_patterns
```

Reply:
It would make it easier to discern which settings belong to which use case

Reply:
then just iterate over the list of specified launchers

Reply:
+1 Most definitely